### PR TITLE
log name/qtype when sending out servfail

### PR DESCRIPTION
**Pull request by Habbie**
Should we check for 'empty qname' to avoid generating another exception? Should we log client? query ID?

### CODE REVIEW

**Comment by pieterlexis** [comment]
Checking for empty might be a good idea

**Comment by ahupowerdns** [comment]
checking for empty is mandatory when logging errors I'm afraid unless we make toString() do it for us.

**Comment by Habbie** [comment]
Updated, please review.

File: pdns/dnsname.hh [hunk 1]
```diff
@@ -52,6 +52,7 @@ public:
   bool operator!=(const DNSName& other) const { return !(*this == other); }
 
   std::string toString(const std::string& separator=".", const bool trailing=true) const;              //!< Our human-friendly, escaped, representation
+  std::string toLogString() const; //!< like plain toString, but returns [empty] on empty names
```
Comment (pieterlexis) [inline pdns/dnsname.hh]: "but returns [empty] on empty names" => "but returns (empty) on empty names"

**Comment by pieterlexis** [comment]
We might need to review all lines that log DNSNames to have consistent output. Other than that, this looks nice

File: pdns/dnsname.hh (hunk 1 above)
Comment (Habbie) [inline pdns/dnsname.hh]: fixed

**Comment by Habbie** [comment]
Yes, review would be nice but is not a holdup for this PR :)

**Comment by pieterlexis** [comment]
indeed, needs a ticket though :)

**Pull request merged**